python-main.js

```diff
@@ -1640,28 +1613,35 @@ function web_editor(config) {
         });
     }
 
+    function closeSerial(keepSession) {
+        console.log("Closing Serial Terminal");
+        $("#repl").empty();
+        $("#repl").hide();
+        $("#request-repl").hide();
+        $("#request-serial").hide();
+        $("#editor-container").show();
+
+        var serialButton = config["translate"]["static-strings"]["buttons"]["command-serial"];
+        $("#command-serial").attr("title", serialButton["title"]);
+        $("#command-serial > .roundlabel").text(serialButton["label"]);
+
+        var daplink = usePartialFlashing ? window.dapwrapper.daplink : window.daplink;
+        daplink.stopSerialRead();
+        daplink.removeAllListeners(DAPjs.DAPLink.EVENT_SERIAL_DATA);
```

Comment:
I think this is the main reason we were seeing double characters on the terminal.
I don't quite have a way to replicate the original issue reported by some users, @microbit-mark do you have any additional info? I couldn't find an issue in this repo.

Reply:
It's mentioned in https://github.com/bbcmicrobit/PythonEditor/issues/253 and originally from 
https://support.microbit.org/a/tickets/24489 However the user says that they are unsure whether they accidentally typed the extra characters or not, so it maybe hard to reproduce.

Reply:
Oh right, this is not the same then.
This issue would have caused all typed characters to appear twice. Before this PR I only saw this type of double characters very sporadically, but this refactor somehow exuberated the issue and this line fixed it, I was hoping it was the same, but it isn't.